.github/workflows/check.yml

```
@@ -32,44 +31,76 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        if [ "$RUNNER_OS" == "Linux" ]; then
+             sudo apt-get update -y && sudo apt-get install fish csh
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+             choco install vcpython27 --yes -f
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+             brew update -vvv && brew install fish tcsh
+        fi
+      shell: bash
     - name: setup python for tox
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
     - name: install tox
       run: python -m pip install tox
+    - uses: actions/checkout@v2
+    - name: pick environment to run
+      run: |
+        print(
+            '::set-env name=TOXENV::' +
+            {'3.8': 'py38', '3.7': 'py37', '3.6': 'py36', '3.5': 'py35', '2.7': 'py27', 'pypy2': 'pypy', 'pypy3': 'pypy3'}['${{ matrix.py }}']
+        )
+      shell: python
     - name: setup python for test ${{ matrix.py }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.py }}
-    - name: setup test suite for ${{ matrix.os }}-${{ matrix.py }}
-      run: tox -vve py --notest
-    - name: run test suite for ${{ matrix.os }}-${{ matrix.py }}
-      run: tox -e py
+    - name: setup test suite
+      run: tox -vv --notest
+    - name: run test suite
+      run: tox
+      env:
+        PYTEST_ADDOPTS: "-vv --durations=20"
+        CI_RUN: 'yes'
+        DIFF_AGAINST: HEAD
+    - name: pick environment to run
+      run: |
+        import os; os.rename('.tox/coverage.{}.xml'.format(os.environ['TOXENV']), '.tox/coverage.xml')
```

Do you have anything against multilining? This works too:
```suggestion
        import os
        os.rename('.tox/coverage.{}.xml'.format(os.environ['TOXENV']), '.tox/coverage.xml')
```

---

In this case, I feel like it does not decreases readability as makes the file longer